crates/turborepo-process/src/child.rs

```diff
@@ -506,71 +484,29 @@ impl Child {
 
     /// Perform a graceful shutdown of the `Child` process.
     pub async fn stop(&mut self) -> Option<ChildExit> {
-        let mut watch = self.exit_channel.clone();
-
-        let fut = async {
-            let child = {
-                let state = self.state.read().await;
-
-                match state.command_channel() {
-                    Some(child) => child,
-                    None => return,
-                }
-            };
-
-            // if this fails, it's because the channel is dropped (toctou)
-            // we can just ignore it
-            child.stop().await.ok();
-        };
-
-        let (_, code) = join! {
```

Comment:
With the lack of state, we no longer need to drive both futures to completion at the same time. Sending the stop command will complete:
 - Once the child manager reads the command from the channel
 - If the child manager exits and closes the command channel

After either of these we can `wait` for the exit code to be reported:
 - If the stop command was read, then the child manager will update the exit status based on the outcome of issuing the command
 - If the stop command was dropped, then the child manager has exited and will have dropped the sender for the exit status channel. If the sender for a watch channel is dropped then `changed()` will exit.